modules/runner/src/main/java/org/apache/ignite/internal/configuration/ClusterConfigurationDefaultsSetter.java

```diff
@@ -0,0 +1,146 @@
+import com.typesafe.config.ConfigFactory;
+import com.typesafe.config.ConfigRenderOptions;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import org.apache.ignite.configuration.ConfigurationDefaultsSetter;
+import org.apache.ignite.configuration.RootKey;
+import org.apache.ignite.configuration.annotation.ConfigurationType;
+import org.apache.ignite.configuration.validation.ConfigurationValidationException;
+import org.apache.ignite.configuration.validation.ValidationIssue;
+import org.apache.ignite.internal.configuration.hocon.HoconConverter;
+import org.apache.ignite.internal.configuration.storage.RuntimeConfigurationStorage;
+import org.apache.ignite.internal.configuration.tree.ConfigurationSource;
+import org.apache.ignite.internal.configuration.tree.ConverterToMapVisitor;
+import org.apache.ignite.internal.configuration.validation.ConfigurationValidatorImpl;
+import org.apache.ignite.internal.manager.IgniteComponent;
+import org.apache.ignite.internal.security.authentication.basic.BasicAuthenticationProviderChange;
+import org.apache.ignite.internal.security.configuration.SecurityConfiguration;
+
+/**
+ * Default configuration setter for cluster configuration.
+ */
+public class ClusterConfigurationDefaultsSetter implements ConfigurationDefaultsSetter, IgniteComponent {
+    private static final String DEFAULT_PROVIDER_NAME = "default";
+
+    private static final String DEFAULT_USERNAME = "ignite";
+
+    private static final String DEFAULT_PASSWORD = "ignite";
+
+    private final ConfigurationManager configurationManager;
+
+    private final ConfigurationValidatorImpl configurationValidator;
+
+    /**
+     * Constructor.
+     *
+     * @param rootKeys Root keys.
+     * @param generator Configuration tree generator.
+     */
+    public ClusterConfigurationDefaultsSetter(
+            Collection<RootKey<?, ?>> rootKeys,
+            ConfigurationTreeGenerator generator
+    ) {
+        configurationValidator = new ConfigurationValidatorImpl(generator, Set.of());
+        configurationManager = new ConfigurationManager(
+                rootKeys,
+                new RuntimeConfigurationStorage(ConfigurationType.DISTRIBUTED),
+                generator,
+                configurationValidator
+        );
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public CompletableFuture<String> setDefaults(String hocon) {
+        ConverterToMapVisitor visitor = ConverterToMapVisitor.builder()
+                .includeInternal(true)
+                .maskSecretValues(false)
+                .build();
+
+        ConfigurationRegistry configurationRegistry = configurationManager.configurationRegistry();
+
+        return validateCfg(hocon)
+                .thenCompose(v -> parseHocon(hocon))
+                .thenCompose(
+                        v -> setSecurityDefaults(configurationRegistry.getConfiguration(SecurityConfiguration.KEY))
```

Comment:
Instead of a hardcode, we may have asked corresponding configuration modules to patch the tree.
Why have you decided to put these defaults into a modules that doesn't own them?